docs/uml/main/ExtEvSimulationClasses.puml

```diff
@@ -1,177 +1,149 @@
-    interface EvDataResponseMessageToExt
-    class AllDepartedEvsRepsonse {
-        - Map<UUID, List<EvMovement>> movements
-    }
-    class ProvideEvcsFreeLots
-    ProvideEvcsFreeLots --|> EvDataResponseMessageToExt
-    AllDepartedEvsRepsonse --|> EvDataResponseMessageToExt
-    ExtEvData -> EvDataMessageFromExt
-    ExtEvData -> EvDataResponseMessageToExt
-    EvMovement -[hidden]-> RequestEvcsFreeLots
-    EvDataMessageFromExt -[hidden]> EvDataResponseMessageToExt
-    EvMovementsMessage -[hidden]> ProvideEvcsFreeLots
-    ProvideEvcsFreeLots -[hidden]> AllDepartedEvsRepsonse
-
-    class ScheduleDataServiceMessage {
-        - ExtEvDataService dataService
-    }
-    EvDataResponseMessageToExt -[hidden]> ScheduleDataServiceMessage
-    ExtEvData -> ScheduleDataServiceMessage
+    node sub1{
+        interface DataResponseMessageToExt
 
+        interface EvDataResponseMessageToExt
 
-    ' LEFT PART
-    interface ExtLinkInterface {
-        + ExtSimulation getExtSimulation()
-        + List<ExtDataSimulation> getExtDataSimulations()
-    }
-    ExtLinkInterface -[hidden]> ExtSimulation
-    ExtLink --|> ExtLinkInterface
+        class ProvideCurrentPrices{
+            + ProvideCurrentPrices()
+        }
 
-    interface ExtTrigger
-    class ActivationMessage {
-        - Long tick
+        class ProvideDepartingEvs{
+            + ProvideDepartingEvs()
+        }
+
+        class ProvideEvcsFreeLots{
+             + ProvideEvcsFreeLots()
+        }
+
+         ProvideEvcsFreeLots --|> EvDataResponseMessageToExt
+         ProvideDepartingEvs --|> EvDataResponseMessageToExt
+         ProvideCurrentPrices --|> EvDataResponseMessageToExt
+         ExtEvData --> EvDataResponseMessageToExt
+         DataResponseMessageToExt --|> EvDataResponseMessageToExt
     }
-    ActivationMessage --|> ExtTrigger
 
-    interface ExtTriggerResponse
-    class CompletionMessage {
-        - Optional<Long> nextActivation
+    node sub3 {
+        class ExtSimAdapterData {
+                    + LinkedBlockingQueue<ExtTrigger> receiveTriggerQueue
+                    - ActorRef extSimAdapter
+                    - String[] mainArgs
+                    + void queueExtMsg(ControlMessageToExt msg)
+                    + void send(ControlResponseMessageFromExt msg)
+                    + String[] getMainArgs()
+        }
+
+        interface ControlMessageToExt
+
+        interface ControlResponseMessageFromExt
+
+        class CompletionMessage
+
+        class ActivationMessage
+
+        ExtSimAdapterData --u> ControlMessageToExt
+        ExtSimAdapterData --u> ControlResponseMessageFromExt
+        CompletionMessage --d|> ControlResponseMessageFromExt
+        ActivationMessage --d|> ControlMessageToExt
+        ExtSimulation -----> ExtSimAdapterData
     }
-    CompletionMessage --|> ExtTriggerResponse
 
-    ExtTrigger -[hidden]> ExtTriggerResponse
-    ExtTrigger <- ExtSimAdapterData
-    ExtTriggerResponse <- ExtSimAdapterData
 }
 
 package simona {
```

Comment:
`EvMovementData` does not exist anymore, that's now `ArrivingEvs` and `DepartingEvsRequest`. Also there's `EvFreeLotsRequest`. Could you update/integrate these? (You can find all of these in the simona project).